packages/peregrine/src/hooks/__tests__/useWindowSize.spec.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { act } from 'react-test-renderer';
+
+import { useEventListener } from '../useEventListener';
+import { WindowSizeContextProvider, useWindowSize } from '../useWindowSize';
+import createTestInstance from '../../util/createTestInstance';
+
+jest.mock('../useEventListener', () => ({
+    useEventListener: jest.fn()
+}));
+
+const Component = () => {
+    useWindowSize();
+
+    return <i />;
+};
+
+test.only('adds an event listener to the window on mount', () => {
+    createTestInstance(
+        <WindowSizeContextProvider>
+            <Component />
+        </WindowSizeContextProvider>
```

Comment:
Excuse my lack of education on hooks but I thought part of the idea was that you didn't have to rely on context like this.

Why doesn't it work for `Component` to just `useWindowSize` directly?

Reply:
Okay nevermind, I just reviewed the actual `useWindowSize.js` file. The reason is because we don't want to add duplicate event listeners, right? 

If so, 👍 but it does seem like this pattern could lead to many top-level context wrappers in the future. I'm cool solving that problem if/when we get there though.

Reply:
> If so, 👍 but it does seem like this pattern could lead to many top-level context wrappers in the future. I'm cool solving that problem if/when we get there though.

Yes, it will ideally lead to that. It's like Redux without the slicing.